File: code/day6.R
library(purrr)

read_answers <- function(file) {
  ans <- readLines(file)
  ids <- cumsum(ans == "")
  map(split(ans, ids), ~ .x[!(.x == "")])
}

ans <- read_answers("data/day6")

# puzzle 1
sum(map_dbl(ans, ~ length(unique(unlist(strsplit(.x, ""))))))

# puzzle 2
sum(map_dbl(ans, ~ sum(table(unlist(strsplit(.x, ""))) == length(.x))))

File: code/day6.jl
function read_answers(file)
	lines = readlines(file)
	answers = [[]]
	for line in lines
		if line == ""
			push!(answers, [])
		else
			push!(last(answers), line)
		end
	end
	answers
end

answers = read_answers("data/day6")

# puzzle 1
[length(unique(split(join(i), ""))) for i in answers] |> sum

# puzzle 2
c = 0
for a in answers
	letters, l = split(join(a), ""), length(a)
	for i in unique(letters)
		global c += count(x -> x == i, letters) == l
	end
end
